### PR TITLE
add french, german and japanese and udpate remaining languages

### PR DESCRIPTION
**Pull request by umarmughal824**
Add french, german and japanese languages support and update remaining languages translations.

### CODE REVIEW

File: poll/translations/config.yaml [hunk 1]
```diff
@@ -13,6 +13,8 @@ locales:
     - pt_BR  # Portuguese (Brazil)
     - ru  # Russian
     - zh_CN  # Chinese (China)
+    - de  # German
+    - ja  # Japaenese
```
Comment (OmarIthawi) [inline poll/translations/config.yaml]: @umarmughal824 Japanese is already there in the code `ja_JP`. Is there any reason why we would like to have two version of Japanese?

File: poll/public/js/translations/ar/textjs.js [hunk 2]
```diff
@@ -4,36 +4,90 @@
                   init: function() {
                     
 
-(function (globals) {
+(function(globals) {
```
Comment (OmarIthawi) [inline poll/public/js/translations/ar/textjs.js]: Why the spacing has been changed? I seems necessary.

Usually those files shouldn't be touched manually.

Comment (umarmughal824) [reply poll/public/js/translations/ar/textjs.js]: Its not touched manually instead its generated in an automated way. I am not sure why this space came.

Comment (OmarIthawi) [reply poll/public/js/translations/ar/textjs.js]: Thanks @umarmughal824, I'll take a look and let you know.

**Comment by OmarIthawi** [comment]
@jcdyer Could you please take a look here? I'm not pretty comfortable to the inconsitencies with the generated `.js` files.

**Comment by OmarIthawi** [comment]
Thanks @jcdyer, I don't think so. We've just recently prepared the i18n pipeline and it has not been updated since.

**Comment by OmarIthawi** [comment]
@umarmughal824 Could you please let me know about the following:

 - How do you install the XBlock in your dev environment?
 - How did you pull the translation from Transifex?

It would be great if you can pull out the commands from your command line history, there's something missing that's causing the scripts to have both the spaces and some incorrect functions.

We need to get the correct pipeline set up in order to move this PR forward. Thanks!

**Comment by antoviaque** [comment]
@OmarIthawi I believe @xitij2000 will be finding a proper reviewer for this PR. Kshitij can you post an update about the plans for this review, here?

**Comment by umarmughal824** [comment]
@OmarIthawi I got the .po files from #Gaurang.
What I did just created the virtual environment for it and install requirements-dev in it.

`pip install xblock-polls/requirements-dev.txt`
`pip install -e xblock-polls`

After than I ran the following commands
`cd xblock-polls`
`python manage.py compilejsi18n` # this command automatically created js file with catalogs having translation files.

**Comment by xitij2000** [comment]
@viadanna Could you have a look here as the reviewer for this PR?
@OmarIthawi Thanks for the help!

**Comment by OmarIthawi** [comment]
@antoviaque Thank you so much! I'll leave this to you :smiley: @xitij2000

File: poll/public/js/translations/fr/textjs.js [hunk 3]
```diff
@@ -4,36 +4,90 @@
...
+  django.catalog = django.catalog || {};
   
-
-  django.interpolate = function (fmt, obj, named) {
-    if (named) {
-      return fmt.replace(/%\(\w+\)s/g, function(match){return String(obj[match.slice(2,-2)])});
-    } else {
-      return fmt.replace(/%s/g, function(match){return String(obj.shift())});
-    }
+  var newcatalog = {
```
Comment (viadanna) [inline poll/public/js/translations/fr/textjs.js]: It seems to be missing some translations, is this expected?
image

**Comment by antoviaque** [comment]
@xitij2000 @viadanna @OmarIthawi  Thanks! :)

File: poll/translations/config.yaml [hunk 4]
```diff
@@ -13,6 +13,7 @@ locales:
     - pt_BR  # Portuguese (Brazil)
     - ru  # Russian
     - zh_CN  # Chinese (China)
+    - de  # German
```
Comment (viadanna) [inline poll/translations/config.yaml]: Shouldn't this be `de_DE` as [edx-platform](https://github.com/edx/edx-platform/blob/master/conf/locale/config.yaml#L22)?

Comment (umarmughal824) [reply poll/translations/config.yaml]: Ok I will update it.

Comment (xitij2000) [reply poll/translations/config.yaml]: @umarmughal824 Insofar as these comments apply to the other translation PRs, please update those as well.

Comment (umarmughal824) [reply poll/translations/config.yaml]: @xitij2000 Yes, I have commented on these other PR's please take a look.

**Comment by umarmughal824** [comment]
@xitij2000 I have update this PR. Can you please review it.

**Comment by xitij2000** [comment]
@umarmughal824 Could you respond to @viadanna's queries above?

**Comment by umarmughal824** [comment]
@xitij2000 I have already update languages extension identical to edx-platform. I have recompiled the .po files still I am getting no space in `(function(globals) {`. I will have to verify each line whether is there any translation being missed or not.

In my opinion we are not missing any becasue we simply call `python manage.py compilejsi18n` so with this whatever is put under translation tag should become the candidate of the .po file.

**Comment by xitij2000** [comment]
@umarmughal824 I am running the exact same command my xblock-poll virtualenv and the files it is generating have the spaces. 

Are your block requirements up to date? I've run  ``make compile_translations`` a couple of times and I get the same result.

**Comment by umarmughal824** [comment]
@xitij2000 Let me send the comit as the result of `make compile_translations`.

**Comment by umarmughal824** [comment]
@xitij2000 I have update the PR with the translations generated from `make compile_translations`. Please take a look.

**Comment by xitij2000** [comment]
@umarmughal824 Here is the commit I get when I compile the translations: https://github.com/open-craft/xblock-poll/commit/1e03f8ec10f0c4200d1948a80df5a679f785ad27

Also, it seems this PR isn't based on the latest master either. Please rebase it.

**Comment by umarmughal824** [comment]
@xitij2000 I have updated it but still test are being failed.

**Comment by xitij2000** [comment]
@umarmughal824 It is a genuine error due to failed validations when working with translations. I think you should start from a clean virtualenv with the latest tools installed and follow the process for translations in the readme for this block.

**Comment by umarmughal824** [comment]
@xitij2000 I have updated my requirements now I got the space for function(globals).

But now I am getting testing failure issue. 
-msgstr ""
+msgstr ""
\ No newline at end of file

Even after new line at end of file. I have tried with both one and two new line at the end of the file. When I run make validate_translations it automatically remove these lines added at the end of the file. That's why the test are kept failing.

I have shared it to you on email as well.

**Comment by xitij2000** [comment]
@umarmughal824 What it's complaining about is that you've added a newline at the beginning and end of the file. Commit the version without the whitespace added and the error will go away.

**Comment by umarmughal824** [comment]
@xitij2000 let me try without newlines.

**Comment by xitij2000** [comment]
@umarmughal824 I have fixed the issue and pushed a commit.

**Comment by xitij2000** [comment]
Did a quick test again, and merging now since it's working.

**Comment by umarmughal824** [comment]
@xitij2000 thanks a lot buddy. I was occupied in some urgent work.

**Pull request merged**